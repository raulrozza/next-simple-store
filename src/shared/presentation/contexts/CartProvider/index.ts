export { default as useAddToCart } from './hooks/useAddToCart';
