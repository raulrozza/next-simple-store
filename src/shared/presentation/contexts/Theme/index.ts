export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
