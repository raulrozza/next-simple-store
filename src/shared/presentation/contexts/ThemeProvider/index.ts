export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
