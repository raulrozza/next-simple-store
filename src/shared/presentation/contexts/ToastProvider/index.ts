export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
