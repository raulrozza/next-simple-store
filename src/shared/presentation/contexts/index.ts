export * from './CartProvider';
export * from './ThemeProvider';
