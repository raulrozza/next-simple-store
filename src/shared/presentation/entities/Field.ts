interface BaseField {
    type: string;
    name: string;
    placeholder?: string;
}

export interface IInput extends BaseField {
    type: 'text';
}
export interface ITextArea extends BaseField {
    type: 'textarea';
}

export type IField = IInput | ITextArea;
