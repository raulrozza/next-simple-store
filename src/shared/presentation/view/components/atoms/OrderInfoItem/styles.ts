import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css`
        padding: ${theme.layout.spacing(3)};

        display: grid;
        grid-template-areas: 'title price' 'list list';
        grid-template-columns: 1fr ${theme.layout.size(9)};
        gap: ${theme.layout.spacing(2)};

        > strong {
            grid-area: title;
            color: ${theme.palette.primary['1000']};
        }

        > span {
            grid-area: price;
            align-self: center;
            justify-self: flex-end;
        }

        ul {
            grid-area: list;
        }
    `}
`;

export const List = styled.ul`
    li {
        display: flex;
        align-items: center;
        justify-content: space-between;
        max-width: ${({ theme }) => theme.layout.size(15)};

        span + span {
            color: ${({ theme }) => theme.palette.primary['800']};
        }
    }
`;
