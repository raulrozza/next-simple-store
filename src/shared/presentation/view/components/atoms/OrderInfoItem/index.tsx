import { FC } from 'react';

import { IOrder } from '@/shared/domain/entities/Order';
import Spacing from '@/shared/presentation/view/components/atoms/Spacing';

import { Container, List } from './styles';

interface OrderInfoItemProps {
    order: IOrder;
}

const OrderInfoItem: FC<OrderInfoItemProps> = ({ order }) => (
    <Container>
        <strong>{order.customer.name}</strong>

        <span>Total: ${order.discountedPrice.toFixed(2)}</span>

        <List>
            <strong>Products ({order.products.length})</strong>

            <Spacing size={2} />

            {order.products.map(entry => (
                <li key={entry.product.id}>
                    <span>
                        x{entry.quantity} {entry.product.name}
                    </span>
                    <span>${entry.product.price.toFixed(2)}</span>
                </li>
            ))}
        </List>
    </Container>
);

export default OrderInfoItem;
