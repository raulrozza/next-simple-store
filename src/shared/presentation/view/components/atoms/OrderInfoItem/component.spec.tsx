import { render, screen } from '@testing-library/react';

import { ICustomer } from '@/shared/domain/entities/Customer';
import { IOrder } from '@/shared/domain/entities/Order';
import { IProduct } from '@/shared/domain/entities/Product';
import { ThemeProvider } from '@/shared/presentation/contexts';

import OrderInfoItem from './index';

describe('OrderInfoItem', () => {
    it('should render the component displaying all the info', () => {
        const view = render(
            <OrderInfoItem
                order={
                    {
                        id: '1',
                        createdAt: new Date(),
                        customer: {
                            name: 'John Doe',
                        } as ICustomer,
                        products: [
                            {
                                product: {
                                    id: 'aushausa',
                                    name: 'Product 1',
                                    price: 10,
                                } as IProduct,
                                quantity: 2,
                            },
                        ],
                        discountedPrice: 20,
                    } as IOrder
                }
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('John Doe')).toBeInTheDocument();
        expect(screen.getByText('Total: $20.00')).toBeInTheDocument();
        expect(screen.getByText('Products (1)')).toBeInTheDocument();
        expect(screen.getByText('x2 Product 1')).toBeInTheDocument();
        expect(screen.getByText('$10.00')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });
});
