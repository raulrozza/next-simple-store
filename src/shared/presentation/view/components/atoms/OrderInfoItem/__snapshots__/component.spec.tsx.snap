// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInfoItem should render the component displaying all the info 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  margin-top: 0.5rem;
}

.c0 {
  padding: 0.75rem;
  display: grid;
  grid-template-areas: 'title price' 'list list';
  grid-template-columns: 1fr 9rem;
  gap: 0.5rem;
}

.c0 > strong {
  grid-area: title;
  color: #244f5c;
}

.c0 > span {
  grid-area: price;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: flex-end;
}

.c0 ul {
  grid-area: list;
}

.c1 li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 15rem;
}

.c1 li span + span {
  color: #4A93A8;
}

<body>
    <div>
      <div
        class="c0"
      >
        <strong>
          John Doe
        </strong>
        <span>
          Total: $
          20.00
        </span>
        <ul
          class="c1"
        >
          <strong>
            Products (
            1
            )
          </strong>
          <div
            class="c2"
            size="2"
          />
          <li>
            <span>
              x
              2
               
              Product 1
            </span>
            <span>
              $
              10.00
            </span>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": .c2 {
  margin-top: 0.5rem;
}

.c0 {
  padding: 0.75rem;
  display: grid;
  grid-template-areas: 'title price' 'list list';
  grid-template-columns: 1fr 9rem;
  gap: 0.5rem;
}

.c0 > strong {
  grid-area: title;
  color: #244f5c;
}

.c0 > span {
  grid-area: price;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: flex-end;
}

.c0 ul {
  grid-area: list;
}

.c1 li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 15rem;
}

.c1 li span + span {
  color: #4A93A8;
}

<div>
    <div
      class="c0"
    >
      <strong>
        John Doe
      </strong>
      <span>
        Total: $
        20.00
      </span>
      <ul
        class="c1"
      >
        <strong>
          Products (
          1
          )
        </strong>
        <div
          class="c2"
          size="2"
        />
        <li>
          <span>
            x
            2
             
            Product 1
          </span>
          <span>
            $
            10.00
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
